File: 3870_Count_Commas_in_Range/sol.java
import java.util.*;

class Solution {
    /**
     * Time:    O(1)
     * Space:   O(1)
     * Strategy: Binary search implementation
     */
    public int solve(int n) {
        return Math.max(0, n-999);
    }
    public static void main(String[] args) {
        Solution sol = new Solution();
        int[] inputs = { 1002, 998 };
        int[] expected = {3, 0};
        System.out.println("Running Tests...");
        for (int i = 0; i < inputs.length; i++) {
            int result = sol.solve(inputs[i]);
            String status = (result == expected[i]) ? "✅ PASS" : "❌ FAIL";
            System.out.printf("%s | Test #%d: Input: %d | Expected: %d | Result: %d%n",
                status, i+1, inputs[i], expected[i], result );
        }
    }
}

File: 3870_Count_Commas_in_Range/sol.ts
/**
 * Time:   O(1)
 * Space:  O(1)
 * Strategy: [Insert Strategy Name]
 */
const solve = (n: number): number => {
    return Math.max(0, n-999);
};
/**
 * Local testing
 */
interface TestCase {
    args: number;
    expected: number;
}
function test(): void {
    const testCases: TestCase[] = [
        { args: 1002, expected: 3 },
        { args: 998, expected: 0 },
    ];
    testCases.forEach(({ args, expected }, index) => {
        const result = solve(args);
        const isPassed = result === expected;
        const status = isPassed ? "✅ PASS" : "❌ FAIL";
        console.log(
            `${status} | Test #${index + 1}: ` +
            `Input: ${JSON.stringify(args)} | ` +
            `Expected: ${expected} | ` +
            `Result: ${result}`
        );
    });
}
test();
export {}; // This forces the file to be treated as a module with its own scope
